pkg/repository/product/productRepositoryImpl.go

```diff
@@ -71,7 +80,10 @@ func (r *productRepositoryImpl) Update(ctx context.Context, id string, product *
 	).Scan(newProduct).Error; err != nil {
 		return nil, errors.New(fmt.Sprintf("failed to update product"))
 	}
-	return newProduct.ToProductEntity(), nil
+
+	productEntity := newProduct.ToProductEntity()
+	r.SetProductSubAttributes(product, sp)
+	return productEntity, nil
 }
 
 func (r *productRepositoryImpl) Delete(ctx context.Context, id string) (*entities.Product, error) {
```

Comment:
**ctx** is not used in this function

Reply:
it last next func in app,  not pass to func next  , i will fix ctx,sp > _,sp